app/controllers/api/v1/packages_controller.rb

```diff
@@ -379,18 +408,68 @@ def package_record
         @package.storage_type = assign_storage_type
         @package.detail = assign_detail if params["package"]["detail_type"].present?
         @package.received_quantity ||= received_quantity
+        @package.offer_id = offer_id
         add_favourite_image if params["package"]["favourite_image_id"]
         @package
       end
 
+      def try_inventorize_package(pkg)
```

Comment:
`@package` is available throughout I think there is no need to pass it.

Reply:
I prefer to write methods this way. Makes them less dependant on state, which in turn makes them moveable, re-usable. As well as more readable since you see the parameter, therefore don't need to guess _what_ will be inventoried.

I'm not a fan of methods called with implicit context generally